enginetest/queries/query_plans.go

```diff
@@ -326,7 +326,7 @@ where
 			"     │       │       └─ org1 (longtext)\n" +
 			"     │       └─ TableAlias(dimension)\n" +
 			"     │           └─ IndexedTableAccess(asset)\n" +
-			"     │               ├─ index: [asset.orgId,asset.assetId]\n" +
+			"     │               ├─ index: [asset.orgId,asset.name,asset.val]\n" +
```

Comment:
we should be picking `(`orgId`,`name`,`assetId`)` here, bit concerned that I missed us changing it to `(orgId,assetID)`

Reply:
It's happening because of the low cardinality of the joined tables.

Basically, when costing non-unique lookups, we assume that each key expression cuts the table in half. If the table is small enough, and if the index has enough key expressions, this can result in a cost lower than the cost of a unique lookup.

As the size of the input tables increases, this effect disappears. I added just four extra rows to the test table and it now chooses the correct index.